File: module-git/src/main/java/com/timecat/module/git/sgit/activities/delegate/actions/RepoAction.java
package com.timecat.module.git.sgit.activities.delegate.actions;

import com.timecat.module.git.sgit.activities.RepoDetailActivity;
import com.timecat.module.git.sgit.database.models.Repo;

public abstract class RepoAction {

    protected Repo mRepo;
    protected RepoDetailActivity mActivity;

    public RepoAction(Repo repo, RepoDetailActivity activity) {
        mRepo = repo;
        mActivity = activity;
    }

    public abstract void execute();
}

File: module-git/src/main/java/com/timecat/module/git/sgit/activities/delegate/actions/RepoAction.kt
package com.timecat.module.git.sgit.activities.delegate.actions

import com.timecat.module.git.sgit.activities.RepoDetailActivity
import com.timecat.module.git.sgit.database.models.Repo

abstract class RepoAction(protected var mRepo: Repo, protected var mActivity: RepoDetailActivity) {
    abstract fun execute()
}